packages/smock/test/smockit/function-manipulation.spec.ts

```diff
@@ -115,6 +115,107 @@ describe('[smock]: function manipulation tests', () => {
     })
   })
 
+  // Currently just an alias for fallback()
+  describe('manipulating receive()', () => {
+    it('should return with no data by default', async () => {
+      const expected = '0x'
+
+      expect(
+        await ethers.provider.call({
+          to: mock.address,
+        })
+      ).to.equal(expected)
+    })
+
+    it('should be able to make a receive function return without any data', async () => {
+      const expected = '0x'
+      mock.smocked.receive.will.return()
+
+      expect(
+        await ethers.provider.call({
+          to: mock.address,
+        })
+      ).to.equal(expected)
+    })
+
+    it('should be able to make a receive function return with data', async () => {
+      const expected = '0x1234123412341234'
+      mock.smocked.receive.will.return.with(expected)
+
+      expect(
+        await ethers.provider.call({
+          to: mock.address,
+        })
+      ).to.equal(expected)
+    })
+
+    it('should be able to make a receive function revert without any data', async () => {
+      mock.smocked.receive.will.revert()
+
+      await expect(
+        ethers.provider.call({
+          to: mock.address,
+        })
+      ).to.be.reverted
+    })
+
+    it('should be able to make a receive function revert with a string', async () => {
+      const expected = 'this is a revert message'
+
+      mock.smocked.receive.will.revert.with(expected)
+
+      await expect(
+        ethers.provider.call({
+          to: mock.address,
+        })
+      ).to.be.revertedWith(expected)
+    })
+
+    it('should be able to make a receive function emit an event', async () => {
+      // TODO
+    })
+
+    it('should be able to change behaviors', async () => {
+      mock.smocked.receive.will.revert()
+
+      await expect(
+        ethers.provider.call({
+          to: mock.address,
+        })
+      ).to.be.reverted
+
+      const expected = '0x'
+      mock.smocked.receive.will.return()
+
+      expect(
+        await ethers.provider.call({
+          to: mock.address,
+        })
+      ).to.equal(expected)
+    })
+
+    describe.skip('resetting the receive function', () => {
```

Comment:
Why is this skipped?

Reply:
`reset` is a function I plan to add eventually(tm) but haven't done yet. So just adding the test as a reminder to myself.

Reply:
you should add a comment here saying that